scripts/mt_to_vcf.py

```diff
@@ -117,18 +115,51 @@ def export_sites_only_vcf(
     Take initial matrix table, convert to sites-only matrix table, then export to vcf
     """
     logger.info('Converting matrix table to sites-only matrix table')
-    final_mt = mt_to_sites_only_mt(mt, n_partitions)
-
-    # export vcf, and return the path
-
+    ht = mt_to_sites_only_ht(mt, n_partitions)
     logger.info(
         f"Exporting sites-only VCF to '{output_path}' to run in the VQSR pipeline"
     )
-    hl.export_vcf(final_mt, output_path)
+    hl.export_vcf(ht, output_path)
     logger.info('Successfully exported sites-only VCF')
 
     return output_path
 
 
+def mt_to_sites_only_ht(mt: hl.MatrixTable, n_partitions: int) -> hl.Table:
+    """
+    Convert matrix table (mt) into sites-only VCF-ready table (ht)
+    :param mt: multi-sample matrix table
+    :param n_partitions: number of partitions for the output table
+    :return: hl.Table
+    """
+
+    mt = _filter_rows_and_add_tags(mt)
+    ht = _create_info_ht(mt, n_partitions=n_partitions)
+    ht = adjust_vcf_incompatible_types(ht)
+    return ht
+
+
+def _filter_rows_and_add_tags(mt: hl.MatrixTable) -> hl.MatrixTable:
+    """Filter rows and add tags"""
+    mt = hl.experimental.densify(mt)
+    # Filter to only non-reference sites
+    mt = mt.filter_rows((hl.len(mt.alleles) > 1) & (hl.agg.any(mt.LGT.is_non_ref())))
```

Comment:
How can `hl.agg.any(mt.LGT.is_non_ref())` evaluate to false when `hl.len(mt.alleles) > 1`?

Reply:
```
mt.filter_rows((hl.len(mt.alleles) > 1) & (~hl.agg.any(mt.LGT.is_non_ref()))).show(3)
locus                        alleles
locus<GRCh38>	array<str>
chr5:81946002	["T","*"]
```